### PR TITLE
Move IExpirationTrigger to Common and rename it to IChangeToken

### PR DESCRIPTION
**Pull request by pranavkm**

### CODE REVIEW

**Comment by pranavkm** [comment]
cc @Tratcher \ @lodejard  \ @davidfowl

**Comment by Tratcher** [comment]
What about StringValues? https://github.com/aspnet/HttpAbstractions/blob/dev/src/Microsoft.Framework.Primitives/StringValues.cs

**Comment by pranavkm** [comment]
OOps. I'll move that too.

File: src/Microsoft.Framework.Primitives/IChangeToken.cs [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+namespace Microsoft.Framework.Caching
+{
+    /// <summary>
+    /// Propagates notifications that a change has occured.
+    /// </summary>
+    public interface IChangeToken
+    {
+        /// <summary>
+        /// Checked each time the key is accessed in the cache.
```
Comment (Tratcher) [inline src/Microsoft.Framework.Primitives/IChangeToken.cs]: These comments are specific to MemoryCache.

**Comment by pranavkm** [comment]
Updated

File: src/Microsoft.Framework.Primitives/IChangeToken.cs [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+{
+    /// <summary>
+    /// Propagates notifications that a change has occured.
+    /// </summary>
+    public interface IChangeToken
+    {
+        /// <summary>
+        /// Gets a value that indicates if a change has occured.
+        /// </summary>
+        bool IsChanged { get; }
```
Comment (Tratcher) [inline src/Microsoft.Framework.Primitives/IChangeToken.cs]: HasChanged?

Comment (davidfowl) [inline src/Microsoft.Framework.Primitives/IChangeToken.cs]: +1

**Comment by davidfowl** [comment]
Change the namespace

**Comment by pranavkm** [comment]
Updated

**Comment by davidfowl** [comment]
:shipit:

**Pull request closed without merging**